components/flex/nz-flex.directive.ts

```diff
@@ -0,0 +1,31 @@
+/**
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/LICENSE
+ */
+
+import { Directive, Input } from '@angular/core';
+
+import { nzAlign, nzDirection, nzFlex, nzGap, nzJustify, nzWrap } from './typings';
+
+@Directive({
+  selector: '[nz-flex],nz-flex',
+  exportAs: 'nzFlex',
+  standalone: true,
+  host: {
+    class: 'ant-flex',
+    '[style.--flex-direction]': `nzDirection`,
+    '[style.--flex-justify]': `nzJustify`,
+    '[style.--flex-align]': `nzAlign`,
+    '[style.--flex-gap]': `nzGap === 'small' ? '8px' : nzGap === 'middle'? '16px': nzGap === 'large' ? '24px' : nzGap + 'px'`,
+    '[style.--flex-wrap]': `nzWrap`,
+    '[style.--flex]': `nzFlex`
+  }
+})
+export class NzFlexDirective {
+  @Input() nzDirection: nzDirection = 'row';
```

Comment:
Let’s fully align ant-design’s flex component parameters.

```ts
@Input() nzVertical: boolean = false;
```

Reply:
A new input named `nzCenter` has been added, which forces the flex container to align its items both vertically and horizontally.

Reply:
We don't need nzCenter yet because ant-design doesn't have one yet.

Reply:
I apologize for the misunderstanding. I thought you wanted to add an input that aligns the items vertically or horizontally. 🤦‍♂️😅